utils/neuralmagic/sparsification_manager.py

```diff
@@ -344,10 +344,18 @@ def qat_active(self, epoch: float) -> bool:
         :param epoch: epoch to check QAT status for
         """
         if self.has_qat_phase:
-            qat_start = min(
-                [mod.start_epoch for mod in self.train_manager.quantization_modifiers]
-            )
-            return qat_start < epoch + 1
+            return self.first_qat_epoch <= epoch + 1
+        else:
+            return False
+
+    def distillation_active(self, epoch: float) -> bool:
+        """
+        Returns true if distillation is turned on for the given epoch
+
+        :param epoch: epoch to check distillation status for
+        """
+        if self.has_distillation_phase:
+            return self.first_distillation_epoch <= epoch + 1
```

Comment:
Why is the +1 needed?

Reply:
So that if distillation is set for epoch 2.5, it gets activated on epoch 2. But it should be `<` rather than `<=`, thanks for bringing attention to it